config.go

```diff
@@ -292,6 +292,19 @@ var (
 	defaultPrunedNodeMaxPeers = 4
 )
 
+// GeneralizedConfig is an interface that defines the functions a config struct
+// must support to be passable in the generalizedConfigLoader function.
+type GeneralizedConfig interface {
+	// GetShowVersion returns the current value for the ShowVersion field
+	GetShowVersion() bool
+
+	// GetLndDir returns the current value for the LndDir field
+	GetLndDir() string
+
+	// GetConfigFile returns the current value for the ConfigFile field
+	GetConfigFile() string
+}
```

Comment:
Let me know if you prefer the approach of this fixup commit that uses an interface instead of passing these functions as params to `generalizedConfigLoader`.

The reason why I didn't squash this commit into the the main commit is that I'd like to get some opinions on which version you prefer, as this approach in this commit is cleaner, but comes with the tradeoff that this adds 3 functions to the main `Config` struct, which is not so nice given that the `cfg` is used throughout `lnd`'s codebase.